src/riscv/offsets.h

```diff
@@ -1,13 +1,7 @@
-#ifdef __linux__
-
 /* Linux-specific definitions: */
 
 /* The RISC-V ucontext has the following structure:
 
    https://github.com/torvalds/linux/blob/44db63d1ad8d71c6932cbe007eb41f31c434d140/arch/riscv/include/uapi/asm/ucontext.h
 */
 #define UC_MCONTEXT_REGS_OFF 176
```

Comment:
Same applies here, and you will actually need to do the correct calculation in `src/riscv/Gstep.c`. At the very least, instead of just removing the `#ifdef __linux__`, add a `__FreeBSD__` check which emits nothing (currently `riscv_handle_signal_frame` just returns `-UNW_EUNSPEC` on non-Linux).